trio/_path.py

```diff
@@ -0,0 +1,145 @@
+from functools import wraps, partial
+import os
+import types
+import pathlib
+
+import trio
+from trio._util import async_wraps
+
+
+__all__ = ['Path']
+
+
+def _forward_factory(cls, attr_name, attr):
+    @wraps(attr)
+    def wrapper(self, *args, **kwargs):
+        attr = getattr(self._wrapped, attr_name)
+        value = attr(*args, **kwargs)
+        if isinstance(value, cls._forwards):
```

Comment:
Oh also: [Codecov says](https://codecov.io/gh/python-trio/trio/pull/180/diff#D4-18) that we never have a test where this branch fails. Probably we should test some method that returns a non-`PurePath`? (Are there any?)

...and for that matter, I guess wrapping `PurePath` objects isn't even necessary, it's only `Path` objects that we need to worry about. (This is an example of a place where I'm finding the generic-ness of this code to add unnecessary confusion; if this was hardcoded to `if isinstance(value, PurePath)` it would be easier to notice :-).)